core/rawdb/chain_freezer.go

```diff
@@ -185,29 +191,54 @@ func (f *chainFreezer) freeze(db ethdb.KeyValueStore) {
 			continue
 		}
 
-		threshold, err := f.freezeThreshold(nfdb)
-		if err != nil {
+		var (
+			frozen    uint64
+			threshold uint64
+			first     uint64 // the first block to freeze
+			last      uint64 // the last block to freeze
+		)
+
+		// Retrieve the freezing threshold.
+		hash := ReadHeadBlockHash(nfdb)
```

Comment:
Mark: it reverts part of the commit: https://github.com/ethereum/go-ethereum/pull/28683/files#diff-08bfad029bc0c2a3735434672787d567c1f1fb4c21135fc40011fa313e36ac43